pkg/cmd/grafana-cli/services/api_client.go

```diff
@@ -0,0 +1,149 @@
+package services
+
+import (
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"runtime"
+
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/models"
+	"github.com/grafana/grafana/pkg/util/errutil"
+	"golang.org/x/xerrors"
+)
+
+type GrafanaComClient struct {
+	retryCount int
+}
+
+func (client *GrafanaComClient) GetPlugin(pluginId, repoUrl string) (models.Plugin, error) {
+	logger.Debugf("getting plugin metadata from: %v pluginId: %v \n", repoUrl, pluginId)
+	body, err := sendRequest(repoUrl, "repo", pluginId)
+
+	if err != nil {
+		logger.Info("Failed to send request: ", err)
+		if err == ErrNotFoundError {
+			return models.Plugin{}, fmt.Errorf("Failed to find requested plugin, check if the plugin_id is correct. error: %v", err)
+		}
+		return models.Plugin{}, fmt.Errorf("Failed to send request. error: %v", err)
+	}
+
+	var data models.Plugin
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		logger.Info("Failed to unmarshal plugin repo response error:", err)
```

Comment:
Not sure this info log is needed. If it is, needs to be changed to `logger.Info("Failed to unmarshal plugin repo response error", "err", err)`